### DISCUSSION TITLE
Docs for NamedPipeEventSelect, NamedPipeEventEnum in KernelBase.dll

### DISCUSSION THREAD

**Discussion by lb90** [microsoft/terminal#18289] [Ideas]
Historically, non-blocking style I/O was supported for Named Pipes via the `PIPE_NOWAIT` flag, however that lacked [readiness notification support](https://devblogs.microsoft.com/oldnewthing/20110114-00/?p=11753). Windows 10 instead comes with two new APIs: `NamedPipeEventSelect`, `NamedPipeEventEnum` (kernelbase.dll) which fill the gap. Would it be possible to document the two new APIs?

**Comment by zadjii-msft** [comment]
Maybe, but this probably isn't the right repo to come asking at. I'm not sure any of us on the Terminal team are familiar with those. @mattwojo any ideas where the documentation for kernelbase apis lives?

**Reply by lb90** [reply]
Thanks @zadjii-msft! Sorry if this isn't the right repo, I opened this discussion as a follow-up to https://github.com/microsoft/terminal/issues/262 which is now closed
